.github/workflows/build_and_deploy.yml

```diff
@@ -2,11 +2,9 @@ name: Build and deploy to AKS cluster
 
 on:
   push:
-    branches:
-    - main
+    branches: [main]
   pull_request:
-    branches:
-    - main
+    types: [labeled, opened, reopened, synchronize]
```

Comment:
pull_request:    
  branches:
    - main
    types: [opened, reopened, synchronize, labeled]

Also line 18 not required as we want build to run, i.e.
if: contains(github.event.pull_request.labels.*.name, 'deploy') || github.event_name != 'pull_request'